discovery/src/main/java/com/ibm/watson/developer_cloud/discovery/v1/model/QueryNoticesOptions.java

```diff
@@ -0,0 +1,416 @@
+   * of possible aggregrations, see the Query reference.
+   */
+  private String aggregation;
+  /** Number of documents to return. */
+  private Long count;
+  /** A comma separated list of the portion of the document hierarchy to return. */
+  private List<String> returnFields;
+  /**
+   * The number of query results to skip at the beginning. For example, if the total number of results that are returned
+   * is 10, and the offset is 8, it returns the last two results.
+   */
+  private Long offset;
+  /**
+   * A comma separated list of fields in the document to sort on. You can optionally specify a sort direction by
+   * prefixing the field with `-` for descending or `+` for ascending. Ascending is the default sort direction if no
+   * prefix is specified.
+   */
+  private String sort;
+  /**
+   * When true a highlight field is returned for each result which contains the fields that match the query with
+   * `<em></em>` tags around the matching query terms. Defaults to false.
+   */
+  private Boolean highlight;
+
+  /**
+   * Builder.
+   */
+  public static class Builder {
+    private String environmentId;
+    private String collectionId;
+    private String filter;
+    private String query;
+    private String naturalLanguageQuery;
+    private Boolean passages;
+    private String aggregation;
+    private Long count;
+    private List<String> returnFields;
+    private Long offset;
+    private String sort;
+    private Boolean highlight;
+
+    private Builder(QueryNoticesOptions queryNoticesOptions) {
+      environmentId = queryNoticesOptions.environmentId;
+      collectionId = queryNoticesOptions.collectionId;
+      filter = queryNoticesOptions.filter;
+      query = queryNoticesOptions.query;
+      naturalLanguageQuery = queryNoticesOptions.naturalLanguageQuery;
+      passages = queryNoticesOptions.passages;
+      aggregation = queryNoticesOptions.aggregation;
+      count = queryNoticesOptions.count;
+      returnFields = queryNoticesOptions.returnFields;
+      offset = queryNoticesOptions.offset;
+      sort = queryNoticesOptions.sort;
+      highlight = queryNoticesOptions.highlight;
+    }
+
+    /**
+     * Instantiates a new builder.
+     */
+    public Builder() {
+    }
+
+    /**
+     * Instantiates a new builder with required properties.
+     *
+     * @param environmentId the environmentId
+     * @param collectionId the collectionId
+     */
+    public Builder(String environmentId, String collectionId) {
+      this.environmentId = environmentId;
+      this.collectionId = collectionId;
+    }
+
+    /**
+     * Builds a QueryNoticesOptions.
+     *
+     * @return the queryNoticesOptions
+     */
+    public QueryNoticesOptions build() {
+      return new QueryNoticesOptions(this);
+    }
+
+    /**
+     * Adds an xreturn to returnFields.
+     *
+     * @param xreturn the new xreturn
+     * @return the QueryNoticesOptions builder
+     */
+    public Builder addXreturn(String xreturn) {
```

Comment:
Yes ... we have a way to rename troublesome fields like `return`.  I will use it here.